device_doctor/lib/src/ios_device.dart

```diff
@@ -0,0 +1,78 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert' show LineSplitter;
+
+import 'package:meta/meta.dart';
+
+import 'device.dart';
+import 'health.dart';
+import 'process_helper.dart';
+
+/// IOS implementation of [DeviceDiscovery].
+///
+/// Discovers available ios devices and chooses one to work with.
+class IosDeviceDiscovery implements DeviceDiscovery {
+  factory IosDeviceDiscovery() {
+    return _instance ??= IosDeviceDiscovery._();
+  }
+
+  IosDeviceDiscovery._();
+
+  @visibleForTesting
+  IosDeviceDiscovery.testing();
+
+  static IosDeviceDiscovery _instance;
+
+  @override
+  Future<List<Device>> discoverDevices({Duration retriesDelayMs = const Duration(seconds: 10)}) async {
+    return LineSplitter.split(await deviceListOutput()).map((String id) => IosDevice(deviceId: id)).toList();
+  }
+
+  Future<String> deviceListOutput() async {
+    return eval('idevice_id', <String>['-l']);
+  }
+
+  @override
+  Future<Map<String, HealthCheckResult>> checkDevices() async {
+    final Map<String, HealthCheckResult> results = <String, HealthCheckResult>{};
+    for (Device device in await discoverDevices()) {
+      results['ios-device-${device.deviceId}'] = HealthCheckResult.success();
+    }
+    return results;
+  }
+
+  @override
+  Future<void> recoverDevices({Duration retriesDelayMs = const Duration(seconds: 10)}) async {
+    for (Device device in await discoverDevices()) {
+      await device.recover();
+    }
+  }
+}
+
+/// iOS device.
+class IosDevice implements Device {
+  const IosDevice({@required this.deviceId});
+
+  @override
+  final String deviceId;
+
+  // The methods below are stubs for now. They will need to be expanded.
+  // We currently do not have a way to lock/unlock iOS devices. So we assume the
+  // devices are already unlocked. For now we'll just keep them at minimum
+  // screen brightness so they don't drain battery too fast.
+
+  @override
+  Future<bool> isAwake() async => true;
+
+  @override
+  Future<bool> isAsleep() async => false;
+
+  @override
+  Future<void> unlock() async {}
```

Comment:
nit: In general, we should throw an unimplemented error on no-ops because we haven't finished the code

Reply:
The following PR is intended to implement corresponding actions. Will leave it there.